src/commands/postDeploy/validateWebSite.ts

```diff
@@ -54,14 +47,10 @@ export async function validateWebSite(deploymentCorrelationId: string, siteTreeI
             }
 
             try {
-                const request: requestUtils.Request = await requestUtils.getDefaultRequest(uri);
-                request.resolveWithFullResponse = true;
-                const response = await requestUtils.sendRequest<IncomingMessage>(request);
-                currentStatusCode = response.statusCode;
+                const response: HttpOperationResponse = await client.sendRequest({ method: 'GET', url });
+                currentStatusCode = response.status;
             } catch (error) {
-                // tslint:disable-next-line:strict-boolean-expressions
-                const response = (<WebError>error).response || {};
-                currentStatusCode = isNumber(response.statusCode) ? response.statusCode : 0;
+                currentStatusCode = error instanceof RestError ? error.statusCode : 0;
```

Comment:
The behavior seems a little different here.  Did we only want to record RestError status codes?

Reply:
I think the difference you're noticing just comes down to "request-promise" vs. "@azure/ms-rest-js". If there is a statusCode in the error, it'll be a `RestError` - that's just how "@azure/ms-rest-js" works. I validated a few sites with successful status codes (2xx) and a few with failed status codes (4xx) and we were working the same